pkgs/by-name/ga/gamdl/package.nix

```
@@ -0,0 +1,137 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  fetchPypi,
+}:
+
+let
+  buildPypiPkg =
+    args:
+    python3Packages.buildPythonApplication (
+      rec {
+        pyproject = true;
+        doCheck = false;
+        propagatedBuildInputs = args.deps ++ (args.optional-deps or [ ]);
+      }
+      // builtins.removeAttrs args [
+        "deps"
+        "optional-deps"
+      ]
+    );
+
+  construct = buildPypiPkg {
+    pname = "construct";
+    version = "2.8.8";
+    src = fetchPypi {
+      pname = "construct";
+      version = "2.8.8";
+      sha256 = "1b84b8147f6fd15bcf64b737c3e8ac5100811ad80c830cb4b2545140511c4157";
+    };
+    nativeBuildInputs = with python3Packages; [ setuptools ];
+    deps = [ ];
+    meta = with lib; {
+      description = "A powerful declarative parser/builder for binary data";
+      homepage = "https://github.com/construct/construct";
+      license = licenses.mit;
+    };
+  };
+
+  protobuf = buildPypiPkg {
+    pname = "protobuf";
+    version = "4.25.1";
+    src = fetchPypi {
+      pname = "protobuf";
+      version = "4.25.1";
+      sha256 = "57d65074b4f5baa4ab5da1605c02be90ac20c8b40fb137d6a8df9f416b0d0ce2";
+    };
+    nativeBuildInputs = with python3Packages; [ setuptools ];
+    deps = [ ];
+    meta = with lib; {
+      description = "Protocol, Buffers";
+      homepage = "https://github.com/protocolbuffers/protobuf";
+      license = licenses.bsd3;
+    };
+  };
+
+  pymp4 = buildPypiPkg {
+    pname = "pymp4";
+    version = "1.4.0";
+    src = fetchPypi {
+      pname = "pymp4";
+      version = "1.4.0";
+      sha256 = "bc9e77732a8a143d34c38aa862a54180716246938e4bf3e07585d19252b77bb5";
+    };
+    nativeBuildInputs = with python3Packages; [ poetry-core ];
+    deps = [ construct ];
+    meta = with lib; {
+      description = "A python library for parsing and manipulating MP4 files";
+      homepage = "https://github.com/beardypig/pymp4";
+      license = licenses.asl20;
+    };
+  };
+
+  pywidevine = buildPypiPkg {
+    pname = "pywidevine";
+    version = "1.8.0";
+    src = fetchPypi {
+      pname = "pywidevine";
+      version = "1.8.0";
+      sha256 = "c14f3fe2864473416b9caa73d9a21251a02d72138e6d54d8c1a3f44b7a6b05c9";
+    };
+    nativeBuildInputs = with python3Packages; [ poetry-core ];
+    deps = with python3Packages; [
+      pycryptodome
+      protobuf
+      requests
+      xmltodict
+      pyyaml
+      unidecode
+      click
+      pymp4
+    ];
+    meta = with lib; {
+      description = "A python implementation of Google's Widevine CDM";
+      homepage = "https://github.com/devine-project/pywidevine";
+      license = licenses.gpl3Only;
+    };
+  };
+
+in
+python3Packages.buildPythonApplication rec {
+  pname = "gamdl";
+  version = "2.5.1";
+
+  src = fetchFromGitHub {
+    owner = "glomatico";
+    repo = "gamdl";
+    rev = "eab33bc02cc7c69e038426f0caa214fb1137bf04";
```

Use the tag instead of a commit revision:
```suggestion
    tag = version;
```